subprojects/core-api/src/main/java/org/gradle/api/artifacts/ComponentDependenciesMetadataDetails.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.artifacts;
+
+import org.gradle.api.Action;
+import org.gradle.api.Incubating;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Describes the dependencies declared in a resolved component's metadata, which typically originate from
+ * a component descriptor (Ivy file, Maven POM). This interface can be used to adjust the dependencies
+ * of a published component via metadata rules (see {@link org.gradle.api.artifacts.dsl.ComponentMetadataHandler}.
+ *
+ * @since 4.4
+ */
+@Incubating
+public interface ComponentDependenciesMetadataDetails extends Collection<ComponentDependencyMetadata> {
+
+    /**
+     * Add a dependency using the string notation: <code><i>group</i>:<i>name</i>:<i>version</i></code>.
+     *
+     * @param dependencyNotation the dependency
+     */
+    void add(String dependencyNotation);
```

Comment:
As an alternative we could name this `create` and avoid the clash with `Collection.add()`.